app/scripts/components/map/index.js

```diff
@@ -168,29 +174,41 @@ export const Map = React.createClass({
     }
 
     // line visibility
-    const hiddenLines = nextProps.draw.hidden;
-
-    this.draw.getAll().features.forEach((feature, i) => {
-      const visible = feature.properties.visibility !== 'none';
-      const featureStatus = feature.properties.status ? feature.properties.status : 'incomplete';
-
-      if (!hiddenLines.length) {
-        if (!visible) this.showLine(feature.id);
-      } else if (!visible && hiddenLines.indexOf(featureStatus) === -1) {
-        this.showLine(feature.id);
-      } else if (visible && hiddenLines.indexOf(featureStatus) > -1) {
-        this.hideLine(feature.id);
+    // const hiddenLines = nextProps.draw.hidden;
+    lineLayers.forEach(layer => {
```

Comment:
it took me a bit to figure out what's going on here, but instead of getting the features from `draw` we're filtering the layers that `draw` adds to the map, yeah? (just making sure I understand this bit)